src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -407,6 +429,95 @@ public override async Task<bool> DrainAsync(int maxDrainBytes)
                     cts?.Dispose();
                 }
             }
+
+            private void ParseHeaderNameValue(ReadOnlySpan<byte> line, HttpResponseHeaders trailers)
```

Comment:
Don't we already have a routine for parsing headers?  We would ideally find a way to reuse that logic rather than duplicating all of it.

Reply:
Agreed. Seems like we should be able to just call back into that routine on the HttpConnection here, except with the TrailingHeaders collection.

Reply:
Actually, it might be more complicated than that. But we should refactor to avoid duplication here.

Reply:
> Don't we already have a routine for parsing headers?

1. In the routine to parse **headers**, we throw immediately when malformed data is found, while in **trailing headers**, we need to be more tolerable/robust, and just skip that line.
2. There is additional logic to eliminate disallowed trailing headers.

I will try if I can refactor the code a bit in next commit.

Reply:
What does "malformed data" mean here? Why do we need to skip instead of just throwing?

Reply:
I was thinking we should be more tolerable (skip, not throw) for invalid trailing headers.

I don't have strong preference here. Will change to throwing to follow `Headers` pattern.